dask_sql/physical/rex/core/call.py

```diff
@@ -115,6 +115,25 @@ def __init__(self, operation: Callable, unary_operation: Callable = None):
 
     def reduce(self, *operands, **kwargs):
         if len(operands) > 1:
+            # Doing math against dates requires a Timedelta
+            # Find all the operands that are of a datetime64 type
+            date_operands = [
+                idx for idx in {0, 1} if pd.api.types.is_datetime64_dtype(operands[idx])
+            ]
+
+            # If there are datetime64 operands we need to make sure that the other operands
+            # in the list are Timedelta for the operation to work.
+            if date_operands:
+                # Operands is a Set, since we are altering it must convert to a List
+                operands = list(operands)
+
+                # Knowing there are datetime types in the operands check for incompatable other types
+                # If found, convert them to Timedelta
+                for idx, operand in enumerate(operands):
+                    # Default to `Day`/`D` since that is what PostgreSQL does
+                    if isinstance(operand, (np.int64, str)):
```

Comment:
In which case would we need to cast a string type to a `timedelta64[D]`?

Reply:
Case I was addressing was some SQL clauses like `TIMESTAMP '2022-06-17'` for example. Where `TIMESTAMP` is accepting a string literal value and not a column.

Reply:
In this case doesn't the `TIMESTAMP '2022-06-17` literal get casted to a np.datetime64 object?

Reply:
Good point. It probably should be but I am for sure seeing `str` instance like that show up here. It is only happening with GPU. I bet somewhere the cast operations are not getting full coverage on GPU specific types.

Reply:
Perfect. Can we open up an issue for tracking that one?

Reply:
Done, https://github.com/dask-contrib/dask-sql/issues/589